### ISSUE TITLE
🐛 [flutter_analytics] setUserProperty with initial value null throws PlatformException

### ISSUE THREAD

**Issue by NebulaFox** [FirebaseExtended/flutterfire#7662]
## Bug report

**Describe the bug**
On `FirebaseAnalytics.instance.setUserProperty(name: "key", value: null)`,  I am getting a `PlatformException`.

```
E/flutter (21473): [ERROR:flutter/lib/ui/ui_dart_state.cc(209)] Unhandled Exception: PlatformException(firebase_analytics, null, null, null)
E/flutter (21473): #0      StandardMethodCodec.decodeEnvelope (package:flutter/src/services/message_codecs.dart:607:7)
E/flutter (21473): #1      MethodChannel._invokeMethod (package:flutter/src/services/platform_channel.dart:167:18)
E/flutter (21473): <asynchronous suspension>
E/flutter (21473): #2      FirebaseAnalytics.setUserProperty (package:firebase_analytics/src/firebase_analytics.dart:210:5)
E/flutter (21473): <asynchronous suspension>
E/flutter (21473): 
E/flutter (21473): [ERROR:flutter/lib/ui/ui_dart_state.cc(209)] Unhandled Exception: PlatformException(firebase_analytics, null, null, null)
E/flutter (21473): #0      StandardMethodCodec.decodeEnvelope (package:flutter/src/services/message_codecs.dart:607:7)
E/flutter (21473): #1      MethodChannel._invokeMethod (package:flutter/src/services/platform_channel.dart:167:18)
E/flutter (21473): <asynchronous suspension>
E/flutter (21473): #2      FirebaseAnalytics.setUserProperty (package:firebase_analytics/src/firebase_analytics.dart:210:5)
E/flutter (21473): <asynchronous suspension>
```

### Steps to reproduce

1. Have a firebase app with analytics
2. After initial launch, `FirebaseAnalytics.instance.setUserProperty(name: "key", value: null)`

The key has not been set to anything

### Expected behavior

Not to throw a `PlatformException`

---

### Flutter doctor

<details><summary>Click To Expand</summary>

```
[✓] Flutter (Channel stable, 2.8.0, on Mac OS X 10.14.6 18G9323 darwin-x64, locale en-GB)
    • Flutter version 2.8.0 at /usr/local/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision cf44000065 (12 days ago), 2021-12-08 14:06:50 -0800
    • Engine revision 40a99c5951
    • Dart version 2.15.0
```

</details>

---

### Flutter dependencies

<details><summary>Click To Expand</summary>

```
Dart SDK 2.15.0
Flutter SDK 2.8.0

dependencies:
- adjust_sdk 4.29.1 [flutter meta]
- async 2.8.2 [collection meta]
- firebase_analytics 9.0.4 [firebase_analytics_platform_interface firebase_analytics_web firebase_core firebase_core_platform_interface flutter]
- firebase_core 1.10.6 [firebase_core_platform_interface firebase_core_web flutter meta]
- firebase_crashlytics 2.4.4 [firebase_core firebase_core_platform_interface firebase_crashlytics_platform_interface flutter stack_trace]
- flutter 0.0.0 [characters collection meta typed_data vector_math sky_engine]
- flutter_secure_storage 5.0.2 [meta flutter flutter_secure_storage_linux flutter_secure_storage_macos flutter_secure_storage_platform_interface flutter_secure_storage_web flutter_secure_storage_windows]
- flutter_svg 1.0.0 [flutter meta path_drawing vector_math xml]
- http 0.13.4 [async http_parser meta path]
- image_picker 0.8.4+4 [flutter flutter_plugin_android_lifecycle image_picker_for_web image_picker_platform_interface]
- intl_phone_number_input 0.7.0+2 [flutter meta libphonenumber_plugin equatable collection]
- logging 1.0.2
- lottie 1.2.1 [archive characters flutter path vector_math]
- mailto 2.0.0
- material_design_icons_flutter 5.0.6295 [flutter]
- provider 6.0.1 [collection flutter nested]
- share 2.0.4 [meta mime flutter]
- sliver_tools 0.2.5 [flutter]
- sprintf 6.0.0
- url_launcher 6.0.17 [flutter meta url_launcher_android url_launcher_ios url_launcher_linux url_launcher_macos url_launcher_platform_interface url_launcher_web url_launcher_windows]

dev dependencies:
- flutter_test 0.0.0 [flutter test_api path fake_async clock stack_trace vector_math async boolean_selector characters charcode collection matcher meta source_span stream_channel string_scanner term_glyph typed_data]
- integration_test 0.0.0 [flutter flutter_driver flutter_test path vm_service archive async boolean_selector characters charcode clock collection crypto fake_async file matcher meta source_span stack_trace stream_channel string_scanner sync_http term_glyph test_api typed_data vector_math webdriver]
- test 1.17.12 [analyzer async boolean_selector collection coverage http_multi_server io js node_preamble package_config path pedantic pool shelf shelf_packages_handler shelf_static shelf_web_socket source_span stack_trace stream_channel typed_data web_socket_channel webkit_inspection_protocol yaml test_api test_core]

dependency overrides:
- libphonenumber 2.0.2 [flutter meta]

transitive dependencies:
- _fe_analyzer_shared 31.0.0 [meta]
- analyzer 2.8.0 [_fe_analyzer_shared cli_util collection convert crypto glob meta package_config path pub_semver source_span watcher yaml]
- archive 3.1.6 [crypto path]
- args 2.3.0
- boolean_selector 2.1.0 [source_span string_scanner]
- characters 1.2.0
- charcode 1.3.1
- cli_util 0.3.5 [meta path]
- clock 1.1.0
- collection 1.15.0
- convert 3.0.1 [typed_data]
- coverage 1.0.3 [args logging package_config path source_maps stack_trace vm_service]
- cross_file 0.3.2 [flutter js meta]
- crypto 3.0.1 [collection typed_data]
- equatable 2.0.3 [collection meta]
- fake_async 1.2.0 [clock collection]
- file 6.1.2 [meta path]
- firebase_analytics_platform_interface 3.0.3 [firebase_core flutter meta plugin_platform_interface]
- firebase_analytics_web 0.4.0+4 [firebase_analytics_platform_interface firebase_core firebase_core_web flutter flutter_web_plugins js]
- firebase_core_platform_interface 4.2.3 [collection flutter meta plugin_platform_interface]
- firebase_core_web 1.5.3 [firebase_core_platform_interface flutter flutter_web_plugins js meta]
- firebase_crashlytics_platform_interface 3.1.11 [collection firebase_core flutter meta plugin_platform_interface]
- flutter_driver 0.0.0 [file flutter flutter_test fuchsia_remote_debug_protocol path meta vm_service webdriver archive async boolean_selector characters charcode clock collection crypto matcher platform process source_span stack_trace stream_channel string_scanner sync_http term_glyph test_api typed_data vector_math]
- flutter_plugin_android_lifecycle 2.0.5 [flutter]
- flutter_secure_storage_linux 1.1.0 [flutter flutter_secure_storage_platform_interface]
- flutter_secure_storage_macos 1.1.0 [flutter flutter_secure_storage_platform_interface]
- flutter_secure_storage_platform_interface 1.0.0 [flutter plugin_platform_interface]
- flutter_secure_storage_web 1.0.2 [flutter flutter_web_plugins flutter_secure_storage_platform_interface js]
- flutter_secure_storage_windows 1.1.2 [flutter flutter_secure_storage_platform_interface]
- flutter_web_plugins 0.0.0 [flutter js characters collection meta typed_data vector_math]
- frontend_server_client 2.1.2 [async path]
- fuchsia_remote_debug_protocol 0.0.0 [process vm_service file meta path platform]
- glob 2.0.2 [async collection file path string_scanner]
- http_multi_server 3.0.1 [async]
- http_parser 4.0.0 [charcode collection source_span string_scanner typed_data]
- image_picker_for_web 2.1.4 [flutter flutter_web_plugins image_picker_platform_interface meta]
- image_picker_platform_interface 2.4.1 [flutter http meta plugin_platform_interface cross_file]
- io 1.0.3 [meta path string_scanner]
- js 0.6.3
- libphonenumber_platform_interface 0.3.1 [flutter plugin_platform_interface]
- libphonenumber_plugin 0.2.3 [flutter flutter_web_plugins libphonenumber_platform_interface libphonenumber_web libphonenumber]
- libphonenumber_web 0.2.0+1 [flutter flutter_web_plugins js libphonenumber_platform_interface]
- matcher 0.12.11 [stack_trace]
- meta 1.7.0
- mime 1.0.1
- nested 1.0.0 [flutter]
- node_preamble 2.0.1
- package_config 2.0.2 [path]
- path 1.8.0
- path_drawing 1.0.0 [vector_math meta path_parsing flutter]
- path_parsing 1.0.0 [vector_math meta]
- pedantic 1.11.1
- petitparser 4.4.0 [meta]
- platform 3.0.2
- plugin_platform_interface 2.0.2 [meta]
- pool 1.5.0 [async stack_trace]
- process 4.2.4 [file path platform]
- pub_semver 2.1.0 [collection meta]
- shelf 1.2.0 [async collection http_parser path stack_trace stream_channel]
- shelf_packages_handler 3.0.0 [path shelf shelf_static]
- shelf_static 1.1.0 [convert http_parser mime path shelf]
- shelf_web_socket 1.0.1 [shelf stream_channel web_socket_channel]
- sky_engine 0.0.99
- source_map_stack_trace 2.1.0 [path stack_trace source_maps]
- source_maps 0.10.10 [source_span]
- source_span 1.8.1 [collection path term_glyph]
- stack_trace 1.10.0 [path]
- stream_channel 2.1.0 [async]
- string_scanner 1.1.0 [charcode source_span]
- sync_http 0.3.0
- term_glyph 1.2.0
- test_api 0.4.3 [async boolean_selector collection meta source_span stack_trace stream_channel string_scanner term_glyph matcher]
- test_core 0.4.2 [analyzer async args boolean_selector collection coverage frontend_server_client glob io meta package_config path pedantic pool source_map_stack_trace source_maps source_span stack_trace stream_channel vm_service yaml matcher test_api]
- typed_data 1.3.0 [collection]
- url_launcher_android 6.0.13 [flutter meta url_launcher_platform_interface]
- url_launcher_ios 6.0.13 [flutter meta url_launcher_platform_interface]
- url_launcher_linux 2.0.2 [flutter]
- url_launcher_macos 2.0.2 [flutter]
- url_launcher_platform_interface 2.0.4 [flutter plugin_platform_interface]
- url_launcher_web 2.0.5 [flutter flutter_web_plugins meta url_launcher_platform_interface]
- url_launcher_windows 2.0.2 [flutter]
- vector_math 2.1.1
- vm_service 7.3.0
- watcher 1.0.1 [async path]
- web_socket_channel 2.1.0 [async crypto stream_channel]
- webdriver 3.0.0 [archive matcher path stack_trace sync_http]
- webkit_inspection_protocol 1.0.0 [logging]
- xml 5.3.1 [collection meta petitparser]
- yaml 3.1.0 [collection source_span string_scanner]
```

</details>

---

**Comment by darshankawar** [comment]
@NebulaFox 
A recent issue was fixed and merged for `setUserId`, https://github.com/FirebaseExtended/flutterfire/pull/7545
But the same need to be done for `setUserProperty` which requires value and name to be `non-null`:

 https://github.com/FirebaseExtended/flutterfire/blob/03751cd24ff65f6484225519709d5e4777a30ca7/packages/firebase_analytics/firebase_analytics/android/src/main/java/io/flutter/plugins/firebase/analytics/FlutterFirebaseAnalyticsPlugin.java#L202

**Comment by kolotum** [comment]
I think #7735 fixes this issue for Android.

On iOS null value is converted to NSNull. It seems to be that the iOS SDK can not handle NSNull correctly. So a minor code change is required on iOS as well:

https://github.com/FirebaseExtended/flutterfire/blob/63e3d0363748b31bc980f8629ffa9a56dc33be53/packages/firebase_analytics/firebase_analytics/ios/Classes/FLTFirebaseAnalyticsPlugin.m#L103

Proposed change:
```objc
[FIRAnalytics setUserPropertyString:[value isKindOfClass:[NSNull class]] ? nil : value forName:name];
```

**Comment by kolotum** [comment]
@Salakar Should I open a new issue to address the fixes for iOS (PR #7810)?
